core-bundle/src/EventListener/DataContainer/ContentCompositionListener.php

```diff
@@ -0,0 +1,285 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\EventListener\DataContainer;
+
+use Contao\Backend;
+use Contao\BackendUser;
+use Contao\CoreBundle\Framework\Adapter;
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\CoreBundle\Routing\Page\PageRegistry;
+use Contao\CoreBundle\Security\ContaoCorePermissions;
+use Contao\CoreBundle\ServiceAnnotation\Callback;
+use Contao\DataContainer;
+use Contao\Image;
+use Contao\LayoutModel;
+use Contao\PageModel;
+use Contao\StringUtil;
+use Doctrine\DBAL\Connection;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Session\Attribute\AttributeBagInterface;
+use Symfony\Component\Security\Core\Security;
+use Symfony\Contracts\Translation\TranslatorInterface;
+use Terminal42\ServiceAnnotationBundle\ServiceAnnotationInterface;
+
+class ContentCompositionListener implements ServiceAnnotationInterface
+{
+    /**
+     * @var ContaoFramework
+     */
+    private $framework;
+
+    /**
+     * @var Security
+     */
+    private $security;
+
+    /**
+     * @var PageRegistry
+     */
+    private $pageRegistry;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    /**
+     * @var Adapter
+     */
+    private $image;
+
+    /**
+     * @var Adapter
+     */
+    private $backend;
+
+    public function __construct(ContaoFramework $framework, Security $security, PageRegistry $pageRegistry, TranslatorInterface $translator, Connection $connection, RequestStack $requestStack)
+    {
+        $this->framework = $framework;
+        $this->security = $security;
+        $this->pageRegistry = $pageRegistry;
+        $this->translator = $translator;
+        $this->connection = $connection;
+        $this->requestStack = $requestStack;
+
+        $this->image = $this->framework->getAdapter(Image::class);
+        $this->backend = $this->framework->getAdapter(Backend::class);
+    }
+
+    /**
+     * @Callback(table="tl_page", target="list.operations.articles.button")
+     */
+    public function renderPageArticlesOperation(array $row, string $href, string $label, string $title, string $icon): string
+    {
+        if (!$this->security->isGranted('contao_user.modules', 'article')) {
```

Comment:
This should be `ContaoCorePermissions::USER_CAN_ACCESS_MODULE`, shouldn't it?

Reply:
Fixed in 0dea19a8229a1f0f3a0c5747349cce228cffda9c